crates/pop-parachains/src/deployer_providers.rs

```diff
@@ -0,0 +1,194 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use std::time::{SystemTime, UNIX_EPOCH};
+use strum::{EnumMessage as _, EnumProperty as _};
+use strum_macros::{AsRefStr, Display, EnumMessage, EnumProperty, EnumString, VariantArray};
+
+/// Supported deployment providers.
+#[derive(
+	AsRefStr,
+	Clone,
+	Debug,
+	Display,
+	EnumMessage,
+	EnumString,
+	EnumProperty,
+	Eq,
+	PartialEq,
+	VariantArray,
+)]
+pub enum DeploymentProvider {
+	/// Polkadot Deployment Portal (PDP). This provider enables seamless deployment of Polkadot
+	/// Native Rollups through the Polkadot Cloud.
+	#[strum(
+		ascii_case_insensitive,
+		serialize = "pdp",
+		message = "Polkadot Deployment Portal",
+		detailed_message = "Effortlessly deploy Polkadot Native Rollups on the Polkadot Cloud",
+		props(
+			BaseURL = "https://staging.deploypolkadot.xyz",
+			CollatorKeysURI = "/api/public/v1/parachains/{para_id}/collators/{chain_name}",
+			DeployURI = "/api/public/v1/parachains/{para_id}/resources",
+		)
+	)]
+	PDP,
+}
+
+impl DeploymentProvider {
+	/// Returns the name of the provider.
+	pub fn name(&self) -> &'static str {
+		self.get_message().unwrap_or_default()
+	}
+
+	/// Returns the detailed description of the provider.
+	pub fn description(&self) -> &'static str {
+		self.get_detailed_message().unwrap_or_default()
+	}
+
+	/// Returns the base URL of the provider.
+	pub fn base_url(&self) -> &'static str {
+		self.get_str("BaseURL").unwrap_or("")
+	}
+
+	/// Constructs the full URI for querying collator keys.
+	///
+	/// # Arguments
+	/// * `relay_chain_name` - The name of the relay chain where deployment will occur.
+	/// * `id` - The ID for which collator keys are being requested.
+	pub fn get_collator_keys_path(&self, relay_chain_name: &str, id: u32) -> String {
+		self.get_str("CollatorKeysURI")
+			.map(|template| {
+				template
+					.replace("{para_id}", &id.to_string())
+					.replace("{chain_name}", relay_chain_name)
+			})
+			.unwrap_or_default()
+	}
+
+	/// Constructs the full URI for deploying a parachain.
+	///
+	/// # Arguments
+	/// * `id` - The ID that is being deployed.
+	pub fn get_deploy_path(&self, id: u32) -> String {
```

Comment:
It seems that we are referring to this as URI instead of path in the doc and within the function.

Wouldn't it make more sense to just call the fn `get_deploy_uri` ?

Reply:
`get_deploy_uri` makes more sense, renamed in https://github.com/r0gue-io/pop-cli/pull/459/commits/10a5b7ef0f12e159ea270ae3e79e2c71f2304e57